src/protocol/command.rs

```diff
@@ -28,8 +44,32 @@ impl Command {
         word.chars().map(escape_char).collect::<Vec<_>>().join("")
     }
 
+    fn executable_with_path(executable: &[u8]) -> PathBuf {
+        which(OsStr::from_bytes(executable)).unwrap_or_else(|| Command::path_from_bytes(executable))
+    }
+
+    fn executable_short_form(executable: &[u8]) -> OsString {
+        let path = Command::path_from_bytes(executable);
+        if !path.is_absolute() {
+            return path.into_os_string();
+        }
+        match path.file_name() {
+            Some(file_name) => {
+                let resolved = Command::executable_with_path(file_name.as_bytes());
+                if resolved == path {
+                    file_name.to_os_string()
+                } else {
+                    path.into_os_string()
+                }
+            }
+            None => path.as_os_str().to_os_string(),
+        }
+    }
```

Comment:
What do you think about pulling out the functions `compare_executables`, `executable_with_path` and `executable_short_form` from `Command`? They could be standalone functions in this module (or in a new module, maybe?).

(And then the tests for these functions could be paired with the functions, i.e. the matching test module could be directly after each function.)

Reply:
Sounds like a good plan. I moved the relevant functions and unit tests to `src/protocol/path_expansion.rs`.